### PR TITLE
fix #3293 Support multi Selection in diff tab

### LINKED ISSUE #3293: Feature: Support multi Selection in diff tab
support opening Beyond compare for more than one file selected at the same time
steps:
select diff tab
select 3 files modified
click on F3
==> beyond compare ( which is defined as default compare tool ) opened with 3 tabs, one tab for each file

### PR DESCRIPTION
**Pull request by hkouki**

### CODE REVIEW

File: GitUI/CommandsDialogs/FormBrowse.cs [hunk 1]
```diff
@@ -2724,7 +2727,7 @@ private void DiffContextMenu_Opening(object sender, System.ComponentModel.Cancel
             var isCombinedDiff = isExcactlyOneItemSelected &&
                 DiffFiles.CombinedDiff.Text == DiffFiles.SelectedItemParent;
             var enabled = isExcactlyOneItemSelected && !isCombinedDiff;
-            openWithDifftoolToolStripMenuItem.Enabled = enabled;
+            //openWithDifftoolToolStripMenuItem.Enabled = enabled;
```
Comment (jbialobr) [inline GitUI/CommandsDialogs/FormBrowse.cs]: Please, remove this line if it is no longer needed.

**Comment by hkouki** [comment]
Fix for #3293

**Comment by jbialobr** [comment]
According to https://github.com/gitextensions/gitextensions/issues/3293#issuecomment-242028568
Can you add a confirmation when an user has selected more than 10 files to be opened in external diff tool?

File: GitUI/GitModuleForm.cs [hunk 2]
```diff
@@ -72,5 +84,30 @@ protected bool ExecuteScriptCommand(int command)
         {
             return Script.ScriptRunner.ExecuteScriptCommand(this, Module, command);
         }
+
+        protected bool checkDiffSelectedItemsLimit(List<GitItemStatus> list)
+        {
+            bool ret = true;
+
+            string diffWarnLimitStr = Module.GetEffectiveSetting("diff.warn.limit");
```
Comment (KindDragon) [inline GitUI/GitModuleForm.cs]: You shouldn't introduce custom .gitconfig settings, but you can add an option to `GitCommands\Settings\AppSettings.cs`

File: GitUI/CommandsDialogs/SettingsDialog/Pages/GitConfigSettingsPage.Designer.cs [hunk 3]
```diff
@@ -555,20 +560,43 @@ private void InitializeComponent()
...
             // 
+            // DiffWarnLabel
+            // 
+            this.DiffWarnLabel.Anchor = System.Windows.Forms.AnchorStyles.Left;
+            this.DiffWarnLabel.AutoSize = true;
+            this.DiffWarnLabel.Location = new System.Drawing.Point(3, 285);
+            this.DiffWarnLabel.Name = "DiffWarnLabel";
+            this.DiffWarnLabel.Size = new System.Drawing.Size(51, 13);
+            this.DiffWarnLabel.TabIndex = 80;
+            this.DiffWarnLabel.Text = "Diff warn";
```
Comment (KindDragon) [inline GitUI/CommandsDialogs/SettingsDialog/Pages/GitConfigSettingsPage.Designer.cs]: Please do not use abbreviations. They are very difficult to translate into other languages

**Comment by hkouki** [comment]
Settings moved to use AppSettings:
gitextensions-settings

gitextensions-warn-message

**Comment by hkouki** [comment]
Branch update and conflit resolved.

**Comment by KindDragon** [comment]
Reopen pull-request then

**Comment by jbialobr** [comment]
The message could tell the exact number of selected files. The dialgo could also include an option to not showing the warning again.

**Comment by jbialobr** [comment]
Wouldn't it be better to put the setting under the Advanced->Confirmations?

**Comment by jbialobr** [comment]
Reopened as https://github.com/gitextensions/gitextensions/pull/3319

**Comment by hkouki** [comment]
Change added and comments added to #3319

**Pull request closed without merging**